src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -534,6 +534,9 @@ public static string GetRelativePath(string directory, string fullPath)
         {
             string relativePath = string.Empty;
 
+            directory = TrimTrailingSeparators(directory);
+            fullPath = TrimTrailingSeparators(fullPath);
+
```

Comment:
this is always terrifying to me.  Is there a reason we have our own GetRelativePath?  I thought this was something the platform provided.  If we write our own, won't we always have the risk of just deviating and needing to continually patch like this?

Reply:
Can you remove this entirely and switch to https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=net-5.0 ?

Reply:
because Path.GetRelativePath doesn't exist in .NET Framework of .NET Standard  I assume? @dotnet/roslyn-compiler

Reply:
That's correct. I don't know if it would be better to use the platform implementation on TFMs where it's available, and fall back to our own implementation otherwise, since if there are bugs with this it may make it harder to reproduce them.

As far as copying the platform implementation, that may help, if the implementation doesn't depend on too much stuff within dotnet/runtime. If it's practical to copy the tests for this method from dotnet/runtime repo that might be helpful too. I consider both to be up to the author's discretion here.

Reply:
I agree that its would be _much_ better if we called the underlying platform apis on .NET Core and newer. The question is whether we want https://github.com/dotnet/roslyn/issues/51602 to make 16.10. we basically have today and tomorrow to get this in if we do.